File: Android_Hospital_Management/Modern_Hospital_Management_System-Backend/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/EmergencyLabOrderService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.EmergencyLabOrderRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.EmergencyLabOrderResponse;

import java.util.List;

public interface EmergencyLabOrderService {

    EmergencyLabOrderResponse create(EmergencyLabOrderRequest request);

    List<EmergencyLabOrderResponse> getByEmergencyPatientId(Long emergencyPatientId);

    List<EmergencyLabOrderResponse> getAll();

    EmergencyLabOrderResponse updateStatus(Long id, String status);

    EmergencyLabOrderResponse updateResult(Long id, String result);

    List<EmergencyLabOrderResponse> getByStatus(String status);

    List<EmergencyLabOrderResponse> getCriticalOrders();
}

File: SPRING/Hospital_Management_system_angular/src/app/services/emergency/emergency-lab-order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { EmergencyLabOrder } from '../../models/emergency';

@Injectable({ providedIn: 'root' })
export class EmergencyLabOrderService {
  private api = environment.apiUrl + 'emergency/lab-orders';

  constructor(private http: HttpClient) {}

  create(order: EmergencyLabOrder): Observable<EmergencyLabOrder> {
    return this.http.post<EmergencyLabOrder>(this.api, order);
  }

  getByEmergencyPatientId(emergencyPatientId: number): Observable<EmergencyLabOrder[]> {
    return this.http.get<EmergencyLabOrder[]>(`${this.api}/patient/${emergencyPatientId}`);
  }

  getAll(): Observable<EmergencyLabOrder[]> {
    return this.http.get<EmergencyLabOrder[]>(this.api);
  }

  updateStatus(id: number, status: string): Observable<EmergencyLabOrder> {
    return this.http.put<EmergencyLabOrder>(`${this.api}/${id}/status`, { status });
  }

  updateResult(id: number, result: string): Observable<EmergencyLabOrder> {
    return this.http.put<EmergencyLabOrder>(`${this.api}/${id}/result`, { result });
  }

  getCriticalOrders(): Observable<EmergencyLabOrder[]> {
    return this.http.get<EmergencyLabOrder[]>(`${this.api}/critical`);
  }
}
